prow/owners/owners.go

```diff
@@ -0,0 +1,460 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package owners
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+)
+
+const (
+	ownersFileName  = "OWNERS"
+	aliasesFileName = "OWNERS_ALIASES"
+	// Github's api uses "" (empty) string as basedir by convention but it's clearer to use "/"
+	baseDirConvention = ""
+)
+
+var dirBlacklist = sets.NewString(".git", "_output")
```

Comment:
Open an issue about making this configurable. We likely want to exclude nested vendor directories downstream.

Reply:
#5156